script/Common.s.sol

```diff
@@ -231,9 +235,11 @@ abstract contract Common is Contracts, Params {
 
     IUniswapV3Pool(_uniV3Pool).initialize(_sqrtPriceX96);
 
-    for (uint256 _i; _i < _cardinality;) {
-      IUniswapV3Pool(_uniV3Pool).increaseObservationCardinalityNext(500);
-      _i += 500;
+    for (uint256 _i = 500; _i < _cardinality; _i += 500) {
+      IUniswapV3Pool(_uniV3Pool).increaseObservationCardinalityNext(uint16(_i));
+    }
+    if (_cardinality % 500 != 0) {
+      IUniswapV3Pool(_uniV3Pool).increaseObservationCardinalityNext(uint16(_cardinality));
     }
```

Comment:
Beware what happens now if `_cardinality` is a multiple of 500 different than zero. I'd correct the `for` statement as follows:

```solidity
for (uint16 _i = 500; _i <= _cardinality; _i += 500) { ... }
```

Also, `_cardinality` is already a `uint16`, so the type casting is unnecessary.